### ISSUE TITLE
`read_pdb`: Trailing whitespace is not removed in column `residue_name`

### ISSUE THREAD

**Issue by kamurani** [Ruibin-Liu/MolDF#32]
I have encountered an issue when reading in a protein PDB file where whitespace is not effectively removed. 

Source: [Q05655](https://alphafold.ebi.ac.uk/files/AF-Q05655-F1-model_v2.pdb)

Using the following code: 

```python
pdb = read_pdb(pdb_file="AF-Q05655-F1-model_v2.pdb", category_names=['_atom_site'])  # We use '_atom_site' here to mirror the mmCIF format and it is the default
atoms_df = pdb['_atom_site']

# Get values for residue_name
list(atoms_df.residue_name.unique())
```
This yields: 
```python
['MET ',
 'ALA ',
 'PRO ',
 'PHE ',
 'LEU ',
 'ARG ',
 'ILE ',
 'ASN ',
 'SER ',
 'TYR ',
 'GLU ',
 'GLY ',
 'GLN ',
 'ASP ',
 'CYS ',
 'VAL ',
 'LYS ',
 'THR ',
 'TRP ',
 'HIS ']
```

This whitespace should be trimmed so that filtering can take place properly. 

Happy to submit a PR for this.

**Comment by Ruibin-Liu** [comment]
Yes, it is annoying when we try to do filtering using `residue_name`. 

I implemented that way simply because of performance consideration since we don't need to trim the extra space for every residue name while reading lines. If we use the `PDBDataFrame` class instead of the raw one to do filtering, it is more convenient and less confusing. The API is like this: https://moldf.readthedocs.io/en/latest/api.html#moldf.pdb_dataframe.PDBDataFrame.residue_names

To avoid the confusion, we can either sacrifice a little bit of reading performance by adding a keyword to the `read_pdb` function so that a `residue_name` column can have the compact version by default, or we can directly return a `PDBDataFrame` object instead of the base class. For the latter, we need to polish the code more before we can confidently use it.

**Comment by kamurani** [comment]
I've just realised the same occurs in other columns, for example: 


```python
>>> atom_df = pdb["_atom_site"]
>>> atom_df.record_name.unique()
array(['ATOM  ', 'TER   ', 'HETATM'], dtype=object)
```

This can be quite confusing especially if one wants to do filtering such as: 

```python 
dff = atom_df[atom_df.record_name == "ATOM"] # this will not work as expected
```
